### PR TITLE
move recentEmojis to ~/.cache/...

### LINKED ISSUE #13: please move recent emoji out of config file
hi!

I would like to configuration-manage (i.e., with ansible) the config for x11-emoji-picker. At present this means that I'll lost my recently used emoji each time I do so.

Could recent emoji be stored in its own dedicated file?

(Might offer a PR for this later, but if it's an easy fix for you and agree, please don't wait for me!)

Thanks!
- Tricia

### PR DESCRIPTION
**Pull request by GaZaTu**
fixes #13

### CODE REVIEW

File: src/EmojiPickerSettings.cpp [hunk 1]
```diff
@@ -85,20 +86,51 @@ EmojiPickerSettings::EmojiPickerSettings(QObject* parent)
...
+public:
+  EmojiPickerCache() : QSettings(path(), QSettings::IniFormat) {}
+
+  ~EmojiPickerCache() {
+    setValue("version", QApplication::applicationVersion());
+  }
+
+private:
+  static QString path() {
+    return QDir::homePath() + "/.cache/" + QApplication::organizationName() + "/" + QApplication::applicationName() + "-mru.ini";
```
Comment (pajlada) [inline src/EmojiPickerSettings.cpp:99]: Use [QStandardPaths::CacheLocation](https://doc.qt.io/qt-5/qstandardpaths.html#StandardLocation-enum) instead

File: README.md [hunk 2]
```diff
@@ -139,7 +139,7 @@ I recommend running it (in the keybind) using the following command instead:
 
 ### Settings
 
-The settings file should be located at `/home/$USER/.config/gazatu.xyz/emoji-picker.ini`.
+The settings file should be located at `$HOME/.config/gazatu.xyz/emoji-picker.ini`.
```
Comment (zneix) [inline README.md:142]:
```suggestion
The settings file should be located at `$XDG_CONFIG_HOME/gazatu.xyz/emoji-picker.ini`.
```

File: README.md [hunk 3]
```diff
@@ -167,6 +166,8 @@ If emojis do not get written into your window: try adding that executable to the
 
 If the dialog is too smol for you: put `QT_SCALE_FACTOR=float` in front of the command. (example: `QT_SCALE_FACTOR=1.25 emoji-picker`)
 
+Some things such as recently used emojis are written to `$HOME/.cache/gazatu.xyz/emoji-picker-mru.ini`
```
Comment (zneix) [inline README.md]:
```suggestion
Some things such as recently used emojis are written to `$XDG_CACHE_HOME/gazatu.xyz/emoji-picker-mru.ini`
```

**Review by zneix — CHANGES_REQUESTED** [review]
Just be a little more specific; with pajlada's suggestion, `QStandardPaths` is XDG-compliant and it will use these variables, and fall back to `$HOME/.config` or `$HOME/.cache` if these are not present.

File: README.md [hunk 4]
```diff
@@ -167,6 +166,8 @@ If emojis do not get written into your window: try adding that executable to the
 
 If the dialog is too smol for you: put `QT_SCALE_FACTOR=float` in front of the command. (example: `QT_SCALE_FACTOR=1.25 emoji-picker`)
 
+Some things such as recently used emojis are written to `$HOME/.cache/gazatu.xyz/emoji-picker/cache.ini`
```
Comment (zneix) [inline README.md:169]: re-review
```suggestion
Some things such as recently used emojis are written to `$XDG_CACHE_HOME/gazatu.xyz/emoji-picker/cache.ini`
```

**Review by zneix — APPROVED** [review]
Very good improvement 👌

**Pull request merged**